src/svd/two-sided/svd.hpp

```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+#include "types.hpp"
+
+/**
+ * Compute SVD of a given n times n matrix using Jacobi rotations.
+ *
+ * The algorithm is described as algorithm 7 in
+ * http://www.cs.utexas.edu/users/inderjit/public_papers/HLA_SVD.pdf
+ * A more detailed and rigorous definition can be found in
+ * http://www.netlib.org/utk/people/JackDongarra/PAPERS/svd-sirev-M111773R.pdf
+ *
+ * @param X      Input matrix of size n times n, given in row-major order.
+ * @param[out] s Output vector of size n, containing the singular values of X
+ *               in order of descending magnitude.
+ * @param[out] B Diagonal matrix of size n times n, in row-major order.
+ *               The diagonal consists of the singular values of X in order
+ *               of descending magnitude.
+ * @param[out] U Orthonormal matrix of size n times n, in row-major order.
+ *               Columns of U are the left singular vectors and the ith column
+ *               corresponds to the ith singular value in s.
+ * @param[out] V Orthonormal matrix of size n times n, in row-major order.
+ *               Columns of V are the right singular vectors and the ith column
+ *               corresponds to the ith singular value in s.
+ * @return       Number of diagonalization iterations to perform.
+ */
+size_t svd(struct matrix_t X, struct vector_t s, struct matrix_t B, struct matrix_t U, struct matrix_t V);
```

Comment:
I think returning only one of `B` and `s` should be enough. As far as I see, we require `B` explicitly in the algorithm, so maybe just return `B`? Or we can make the interfaces more similar, and just return `s`. Which one do you think is better?

Reply:
I also thought about this. I tried it to do the function as generic as possible but maybe it is better to change it to return only ```s``` to share a similar interface with one-sided.